workflows: improve logging and search criteria for auto-merge workflow

rafiss opened the pull request:
### workflows: log when auto-merge workflow has nothing to do

### workflows: broaden criteria for auto-merge workflow

Previously, the search excluded PRs made by the blathers bot account as
well as PRs that didn't have the blathers-backport label.

In practice, this meant it only found PRs that were made by blathers
_and also_ only if the original PR author had set up oauth to allow
blathers to automatically backport it using their account.

Now, the search will find any test only backport PR, regardless of
author.

Epic: None
Release note: None

rafiss committed:
workflows: log when auto-merge workflow has nothing to do

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/152686)
<!-- Reviewable:end -->

celiala reviewed (approved):
LGTM

rafiss committed:
workflows: broaden criteria for auto-merge workflow

Previously, the search excluded PRs made by the blathers bot account as
well as PRs that didn't have the blathers-backport label.

In practice, this meant it only found PRs that were made by blathers
_and also_ only if the original PR author had set up oauth to allow
blathers to automatically backport it using their account.

Now, the search will find any test only backport PR, regardless of
author.

Release note: None

rafiss commented:
tftr!

bors r+
